src/rgw/rgw_op.cc

```diff
@@ -1027,13 +1027,15 @@ int RGWGetObj::handle_slo_manifest(bufferlist& bl)
 
   for (const auto& entry : slo_info.entries) {
     const string& path = entry.path;
-    const size_t pos = path.find('/', 1); /* skip first / */
-    if (pos == string::npos) {
+    /* If the path starts with slashes, strip them all. */
+    const size_t pos_init = path.find_first_not_of('/');
+    const size_t pos_sep = path.find('/', pos_init);
```

Comment:
@yehudasa: not sure whether I understood you correctly. I think that passing `std::string::npos` to `find` is safe - unlike in case of `substr` when we could get `std::out_of_range` exception. [The documentation]() says about the `pos` argument of `find`:
> If this is greater than the string length, the function never finds matches.

In C++11 the `char` variant also has the no-throw guarantee:
> character (4)	 `size_t find (char c, size_t pos = 0) const noexcept;`